pkg/util/test/util.go

```diff
@@ -576,6 +594,167 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
+			go func(p v1.Pod) {
+				defer wg.Done()
+				f := func() (interface{}, bool, error) {
+					logrus.Debugf("NodeWiper pod [%s] ; State: [%s]", p.Name, p.Status.Phase)
+					if !coreops.Instance().IsPodRunning(p) {
+						return nil, true, fmt.Errorf("NodeWiper pod [%s] is not running yet", p.Name)
+					}
+					return nil, false, nil
+				}
+				if _, err := task.DoRetryWithTimeout(f, timeout, interval/2); err != nil {
+					logrus.Errorf("NodeWiper pod [%s] timed out to be ready", p.Name)
+					podsNotReady = append(podsNotReady, p.Name)
+				} else {
+					logrus.Infof("NodeWiper pod [%s] is Running!", p.Name)
+					podsReady = append(podsReady, p.Name)
+				}
+			}(pod)
+		}
+		wg.Wait()
+
+		if len(podsReady) != len(expectedPxNodeNameList) {
+			return fmt.Errorf("failed to get following NodeWiper pods ready on storageNodes: [%s]", podsNotReady)
+		}
+		logrus.Debugf("All NodeWiper pods are ready on storageNodes: [%s]", podsReady)
+	}
+
+	// Validate images are correct
+	opVersion, err := GetPxOperatorVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get operator version, Err: %v", err)
+	}
+	k8sVersion, err := GetK8SVersion()
+	if err != nil {
+		return fmt.Errorf("failed to get k8s version, Err: %v", err)
+	}
+
+	// check if px-versions config-map exists
+	var pxImageList map[string]string
+	config, _ := coreops.Instance().GetConfigMap(pxVersionsConfigmapName, cluster.Namespace)
+	if config != nil {
+		logrus.Debugf("px-versions config-map found, fetching images from config-map")
+		pxImageList, err = GetImagesFromConfigMap(config)
+		if err != nil {
+			return fmt.Errorf("failed to get images from config-map: [%v]", err)
+		}
+	} else {
+		pxSpecURL := reconstructSpecURL(cluster)
+		pxImageList, err = GetImagesFromVersionURL(pxSpecURL, k8sVersion)
+		if err != nil {
+			return fmt.Errorf("failed to get images from version URL: [%v]", err)
+		}
+	}
+
+	pxVersion := GetPortworxVersion(cluster)
+
+	var nodeWiperImage string
+	if opVersion.GreaterThanOrEqual(opVer23_10_2) {
+		if pxVersion.GreaterThanOrEqual(pxVer3_1) {
+			pxImage := cluster.Spec.Image
+			pxVersionStr := strings.Split(pxImage, ":")[len(strings.Split(pxImage, ":"))-1]
+			nodeWiperImage = fmt.Sprintf("portworx/oci-monitor:%s", pxVersionStr)
+		} else {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		}
+	} else if opVersion.LessThan(opVer23_10_2) {
+		if pxVersion.LessThan(pxVer3_1) {
+			nodeWiperImage = pxImageList["nodeWiper"]
+		} else {
+			nodeWiperImage = defaultNodeWiperImage
+		}
+	}
+	logrus.Debugf("Expected nodeWiper image: [%s]", nodeWiperImage)
+	err = validateContainerImageInsidePods(cluster, nodeWiperImage, "px-node-wiper", pods)
+	if err != nil {
+		return fmt.Errorf("failed to validate NodeWiper image: [%v]", err)
+	}
+	logrus.Debugf("NodeWiper images match!")
+
+	return nil
+}
+
+func GetImagesFromConfigMap(config *v1.ConfigMap) (map[string]string, error) {
+	pxImageList := map[string]string{}
+	versionData, ok := config.Data["versions"]
```

Comment:
this is how configmap looks like
```
npopov@npopov--MacBookPro18 ~/nikolay-ocp-1 :~$ kubectl -n portworx get cm px-versions -o yaml
apiVersion: v1
data:
  versions.yaml: |
    version: 3.1.1
    components:
      stork: openstorage/stork:23.11.1
      autopilot: portworx/autopilot:1.3.14
      csiNodeDriverRegistrar: registry.k8s.io/sig-storage/csi-node-driver-registrar:v2.9.0
      csiProvisioner: registry.k8s.io/sig-storage/csi-provisioner:v3.6.1
      csiAttacher: docker.io/openstorage/csi-attacher:v1.2.1-1
      csiResizer: registry.k8s.io/sig-storage/csi-resizer:v1.9.1
      csiSnapshotter: registry.k8s.io/sig-storage/csi-snapshotter:v6.3.1
      csiSnapshotController: registry.k8s.io/sig-storage/snapshot-controller:v6.3.1
      prometheus: quay.io/prometheus/prometheus:v2.48.1
      prometheusOperator: quay.io/prometheus-operator/prometheus-operator:v0.70.0
      prometheusConfigReloader: quay.io/prometheus-operator/prometheus-config-reloader:v0.70.0
      alertManager: quay.io/prometheus/alertmanager:v0.26.0
      telemetry: purestorage/ccm-go:1.2.2
      pxLibUpdate: portworx/px-lib:pxfslibs-3.1.0.0-mver12-update
      metricsCollector: purestorage/realtime-metrics:1.0.23
      telemetryProxy: purestorage/telemetry-envoy:1.1.11
      logUploader: purestorage/log-upload:px-1.1.8
      dynamicPlugin: portworx/portworx-dynamic-plugin:1.1.0
      dynamicPluginProxy: nginxinc/nginx-unprivileged:1.25
kind: ConfigMap
metadata:
  creationTimestamp: "2024-05-02T18:25:45Z"
  name: px-versions
  namespace: portworx
  resourceVersion: "27319378"
  uid: 4c4dbd88-875a-467b-92f5-766a9557e1da
```
I don't think this is going to work, I also see failures in the CI 
```
07:25:09  time="2024-05-02T14:23:09Z" level=debug msg="px-versions config-map found, fetching images from config-map"
07:25:09  time="2024-05-02T14:23:09Z" level=error msg="Failed to validate node wiper: failed to get images from config-map: [versions key not found in config-map]"

```

Reply:
See https://docs.portworx.com/portworx-enterprise/install-portworx/kubernetes/airgapped/air-gapped-baremetal

Reply:
Ok looks like Operator has a logic to get any first data key and use it, if correct one has not been passed https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/manifest/configmap.go#L48

Reply:
Updated the logic for `GetImagesFromConfigMap` to ignore the key if its not 'versions'